pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -41,20 +41,23 @@ var (
 )
 
 // IsReadyToReconcile determines if a controller is ready to reconcile or not
-func IsReadyToReconcile(client client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName) (cephv1.ClusterSpec, bool, reconcile.Result) {
+func IsReadyToReconcile(client client.Client, clustercontext *clusterd.Context, namespacedName types.NamespacedName) (cephv1.ClusterSpec, bool, bool, reconcile.Result) {
 	namespacedName.Name = namespacedName.Namespace
+	cephClusterExists := true
 
 	// Running ceph commands won't work and the controller will keep re-queuing so I believe it's fine not to check
 	// Make sure a CephCluster exists before doing anything
 	cephCluster := &cephv1.CephCluster{}
 	err := client.Get(context.TODO(), namespacedName, cephCluster)
 	if err != nil {
 		if kerrors.IsNotFound(err) {
+			cephClusterExists = false
 			logger.Errorf("CephCluster resource %q not found in namespace %q", namespacedName.Name, namespacedName.Namespace)
-			return cephv1.ClusterSpec{}, false, ImmediateRetryResult
+			return cephv1.ClusterSpec{}, false, cephClusterExists, ImmediateRetryResult
 		} else if err != nil {
+			cephClusterExists = false
```

Comment:
If the error is not the `notFound` error, the cluster CR might actually exist. So if this is triggering removal of the finalizer I don't think we want to set it to false here. This is just likely an intermittent error.